### DISCUSSION TITLE
TypeError when trying to open outstatic

### DISCUSSION THREAD

**Discussion by RyanxWolton** [avitorio/outstatic#172] [Q&A] [answered]
Using: 
outstatic 1.0.5
react 18.2.0
next 14.0.4 (I've also tried 13.5.4 as per the example in this repo)

I've followed the guide on adding outstatic to Next.js, but when navigating to /outstatic I'm confronted with the following error. Am I missing something? I've not even been able to get to the point of it complaining about environment variables. 
 
 ○ Compiling /outstatic/[[...ost]]/page ...
 ✓ Compiled /outstatic/[[...ost]]/page in 8.9s (5519 modules)
 ○ Compiling /outstatic/[[...ost]]/route ...
 ✓ Compiled /outstatic/[[...ost]]/route in 3.9s (2840 modules)
 ⨯ TypeError: Cannot read properties of undefined (reading '0')
    at eval (webpack-internal:///(rsc)/./node_modules/outstatic/dist/index.js:603:24)
    at Generator.next (<anonymous>)
    at eval (webpack-internal:///(rsc)/./node_modules/outstatic/dist/index.js:47:31)
    at new Promise (<anonymous>)
    at o (webpack-internal:///(rsc)/./node_modules/outstatic/dist/index.js:33:20)
    at GET (webpack-internal:///(rsc)/./node_modules/outstatic/dist/index.js:598:18)

**Comment by avitorio** [comment]
Hi @RyanxWolton 
Thanks for flagging this. Is this happening on a clean Next.js install? If not, would it be possible to provide me a link to a repo with the same setup or give me access to the repo where this is happening? Thanks.

**Reply by RyanxWolton** [reply]
Hey @avitorio,

Yep tried a clean install and the result is the same. If you're unable to reproduce it let me know and I'll invite you to my repo.

**Reply by avitorio** [reply]
Hi @RyanxWolton 
I just created a new Next.js install following the docs:

https://nextjs.org/docs/getting-started/installation
https://outstatic.com/docs/getting-started#adding-outstatic-to-a-nextjs-website

And it is asking me the env vars.

Can you add me to your repo? What are you using to compile/run the project? yarn, npm, pnpm, bun?

**Reply by RyanxWolton** [reply]
I've invited you to collab, I'm using yarn

**Reply by avitorio** [reply, marked as answer]
Hey @RyanxWolton,
Found the error. You placed the `route.ts` file inside the wrong folder.
It should be under `/app/api/outstatic/[[...ost]]/route.ts` You forgot the `/api/` part.
It should work now.

**Reply by RyanxWolton** [reply]
@avitorio ah man, that's my bad! I must have misread the second part. Thank you for taking the time to check, very much appreciated.
